tests/unit/django/test_smtp.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Flask charm smtp relation unit tests."""
+
+import ops
+import pytest
+from ops.testing import Harness
+
+from .constants import DEFAULT_LAYER, DJANGO_CONTAINER_NAME
+
+SMTP_DATA = {
+    "auth_type": "none",
+    "domain": "example.com",
+    "host": "test-ip",
+    "port": "1025",
+    "skip_ssl_verify": "False",
+    "transport_security": "none",
+}
+
+
+def test_smtp_relation(harness: Harness, request: pytest.FixtureRequest):
```

Comment:
Could we do these as parametrised tests for each of the frameworks? Rather than repeat them a lot?